### PR TITLE
Add Gamma sign in and gamma service

### PR DESCRIPTION
**Pull request by Oscariremma**
Adds the ability to login via gamma OAUTH and also provides a service to interact with the gamma API.

### CODE REVIEW

File: src/auth/auth.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { NextAuthOptions } from 'next-auth';
+import GammaProvider from '@/auth/GammaProvider';
+
+export const authConfig: NextAuthOptions = {
+  providers: [
+    GammaProvider({
+      authorizationURL: 'http://localhost:8081/api/oauth/authorize',
```
Comment (GAsplund) [inline src/auth/auth.ts]: Shouldn't these be environment variables?

**Review by GAsplund — CHANGES_REQUESTED** [review]
Looks great, but check my annotation.

File: src/auth/auth.ts (hunk 1 above)
Comment (Oscariremma) [reply src/auth/auth.ts]: You are correct!

**Comment by GAsplund** [comment]
Also hijacking this PR to note that our documentation (`docker-compose.dev.yml`) does not match reality (`docker-compose-dev.yml`). I'm thinking since you're already editing that file, maybe you could rename it? (Alternatively, we could do it the other way by changing the documentation, it's up to you)

**Comment by Oscariremma** [comment]
> Also hijacking this PR to note that our documentation (`docker-compose.dev.yml`) does not match reality (`docker-compose-dev.yml`). I'm thinking since you're already editing that file, maybe you could rename it? (Alternatively, we could do it the other way by changing the documentation, it's up to you)

Fixed it by renaming the file

**Review by GAsplund — APPROVED** [review]
LGTM!

**Pull request merged**